Add floatValue to ExecuTorch value

f-meloni opened the pull request:
Summary:
I think we can have an utility to return floats back.
I would not create a new tag, but just use the double value and return it as float if the user needs float

Differential Revision: D74603334




cc @mergennachin @byjlw

f-meloni commented:
@pytorchbot label "module: user experience"

shoumikhin commented on extension/apple/ExecuTorch/Exported/ExecuTorchValue.h:
```diff
@@ -40,6 +40,8 @@ typedef NSInteger ExecuTorchIntegerValue
     NS_SWIFT_BRIDGED_TYPEDEF NS_SWIFT_NAME(IntegerValue);
 typedef double ExecuTorchDoubleValue
     NS_SWIFT_BRIDGED_TYPEDEF NS_SWIFT_NAME(DoubleValue);
+    typedef double ExecuTorchFloatValue
```
```suggestion
typedef float ExecuTorchFloatValue
```

shoumikhin replied:
Should this be float?

shoumikhin commented on extension/apple/ExecuTorch/Exported/ExecuTorchValue.h:
```diff
@@ -149,6 +158,13 @@ __attribute__((deprecated("This API is experimental.")))
+/**
+ * Returns YES if the value is a float.
+ *
+ * @return A BOOL indicating whether the value is a double.
```
```suggestion
 * @return A BOOL indicating whether the value is a float.
```

shoumikhin commented on extension/apple/ExecuTorch/Exported/ExecuTorchValue.mm:
```diff
@@ -118,6 +123,10 @@ - (BOOL)isDouble {
 }
 
+- (BOOL)isFloat {
+  return _tag == ExecuTorchValueTagDouble;
```
```suggestion
  // EValue does not have a separate tag for float.
  return _tag == ExecuTorchValueTagDouble;
```

shoumikhin commented on extension/apple/ExecuTorch/Exported/ExecuTorchValue.h:
```diff
@@ -100,6 +102,13 @@ __attribute__((deprecated("This API is experimental.")))
 @property(nonatomic, readonly) ExecuTorchDoubleValue doubleValue NS_SWIFT_NAME(double);
 
+/**
+ * The integer value if the tag is ExecuTorchValueTagFloat.
+ *
+ * @return An NSInteger representing the float value.
```
```suggestion
 * The float value if the tag is ExecuTorchValueTagDouble.
 *
 * @return A float representing the float value.
```

shoumikhin commented on extension/apple/ExecuTorch/Exported/ExecuTorchValue.mm:
```diff
@@ -88,6 +88,12 @@ - (ExecuTorchDoubleValue)doubleValue {
 
+- (ExecuTorchFloatValue)floatValue {
+  ET_CHECK(self.isFloat);
+  // EValue does not have a separate tag for float.
```
This comment belongs to isFloat implementation?

f-meloni commented:
@pytorchbot label "release notes: api"

Commit:
Add floatValue to ExecuTorch value (#10823)

Summary:

I think we can have an utility to return floats back.
I would not create a new tag, but just use the double value and return it as float if the user needs float

Reviewed By: shoumikhin

Differential Revision: D74603334
